test/Concurrency/async_let_isolation.swift

```diff
@@ -5,7 +5,7 @@ actor class MyActor {
   let immutable: Int = 17
   var text: [String] = []
 
-  func synchronous() -> String { text.first ?? "nothing" } // expected-note 2 {{only asynchronous methods can be used outside the actor instance; do you want to add 'async'?}}
+  func synchronous() -> String { text.first ?? "nothing" } // expected-note 2 {{calls to instance method 'synchronous()' from outside of its actor context are implicitly asynchronous}}
   func asynchronous() async -> String { synchronous() }
 
   func testAsyncLetIsolation() async {
```

Comment:
I'm a bit wary about whether the current implementation of implicitly async is interacting correctly with `async let`. Actor-isolation checking rejects:

```swift
async let x = self.synchronous()
```

only for executing in a concurrent context, which is good, but should we still emit the note about implicitly async? It's actually *not* implicitly async there, since no await (seems) to be needed.

Reply:
The initializer of an `async let` runs concurrently with the rest of the function body, as if it were in an escaping closure, so it is correct to diagnose (before you PR) and connect to promote to async (with your PR).

Reply:
Gotcha. In that case it looks like this is good to go. I've added a bit more to the async-let test to make sure it's treated correctly.